activesupport/lib/active_support/core_ext/object/deep_dup.rb

```diff
@@ -43,7 +43,7 @@ class Hash
   def deep_dup
     hash = dup
     each_pair do |key, value|
-      if key.frozen? && ::String === key
+      if ::Symbol === key || (::String === key && key.frozen?)
```

Comment:
I have a slight preference for flipping these to keep the string case as fast as possible, since we call `deep_dup` on string keys every time we [instantiate any model](https://github.com/rails/rails/blob/f2019462f9654bad110d9d3c26b54c2ac30434cd/activerecord/lib/active_record/core.rb#L343).

```suggestion
      if (::String === key && key.frozen?) ||  ::Symbol === key
```

I think I ran a benchmark that showed this being a bit faster for strings, but probably best to confirm that.

Reply:
works for me.  just switched them and reran benchmarks.  perf for string keys is the same as before, symbols see ~27% improvement